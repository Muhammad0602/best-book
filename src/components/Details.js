import React from 'react';
import { useSelector } from 'react-redux';
import { useNavigate, useParams } from 'react-router-dom';

function Details() {
  const navigate = useNavigate();
  const { number } = useParams();
  const id = parseInt(number, 10);
  const { chapters } = useSelector((store) => store.chapters);
  const surah = chapters[id - 1];
  return (
    <div>
      <button onClick={() => navigate('/')} type="button">back</button>
      <h2>Here you can read ayahs of a specific surah</h2>
      <article className="ayahs">
        {surah.ayahs.map((ayah) => (
          <p className="ayah" key={ayah.number}>{ayah.text}</p>
        ))}
      </article>
    </div>
  );
}

export default Details;
