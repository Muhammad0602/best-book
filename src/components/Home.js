import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getChapters } from '../redux/chapters/chaptersSlice';

function Home() {
  const {
    chapters, search, isLoading, error,
  } = useSelector((store) => store.chapters);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getChapters());
  }, [dispatch]);

  if (isLoading) {
    return (
      <div className="loading">
        <div className="spinner" />
      </div>
    );
  }

  if (error) {
    return (
      <div className="error-container">
        <h2>Oopps somethings went wrong.PLease try again!</h2>
        <p>{error}</p>
      </div>
    );
  }

  return (
    <div className="list-surahs">
      {chapters.length > 1
        ? chapters.filter((sura) => (search
          ? sura.englishName.toLowerCase().includes(search) || sura.name.includes(search)
           || parseInt(sura.number, 10) === parseInt(search, 10) : sura))
          .map((surah) => (
            <Link
              to={`/Details/${surah.number}`}
              key={surah.number}
              className="surah-container"
            >
              <h2>{surah.name}</h2>
              <p>{surah.englishName}</p>
              <h4 className="surah-number">{surah.number}</h4>
            </Link>
          ))
        : (
          <Link
            to={`/Details/${chapters.number}`}
            key={chapters.number}
            className="surah-container"
          >
            <h2>{chapters.name}</h2>
            <p>{chapters.englishName}</p>
            <h4 className="surah-number">{chapters.number}</h4>
          </Link>
        )}
    </div>
  );
}

export default Home;
