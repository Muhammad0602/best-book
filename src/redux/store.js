import { configureStore } from '@reduxjs/toolkit';
import chaptersReducer from './chapters/chaptersSlice';
import englishReducer from './english/englishSlice';

const store = configureStore({
  reducer: {
    chapters: chaptersReducer,
    english: englishReducer,
  },
});

export default store;
