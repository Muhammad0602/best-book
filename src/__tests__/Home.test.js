import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import Home from '../components/Home';
import Details from '../components/Details';

describe('Renders without crashing', () => {
  it('should render correctly', () => {
    const number = 4;
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path={`/Details/${number}`} element={<Details />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
