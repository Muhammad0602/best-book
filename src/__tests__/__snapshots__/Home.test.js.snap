// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Home should render correctly 1`] = `
<div
  className="list-surahs"
/>
`;

exports[`Renders without crashing should render correctly 1`] = `null`;
